### DISCUSSION TITLE
Let's encrypt certificate not updating

### DISCUSSION THREAD

**Discussion by trezn0** [teslamate-org/teslamate#3426] [General]
I have tried to update the let's encrypt certificate to no avail. I did as the let's encrypt site says about forced update of certificates, but this results in this acme.json file:

`{
  "tmhttpchallenge": {
    "Account": {
      "Email": "redacted",
      "Registration": {
        "body": {
          "status": "valid",
          "contact": [
            "mailto:redacted"
          ]
        },
        "uri": "https://acme-v02.api.letsencrypt.org/acme/acct/1413612826"
      },
      "PrivateKey": "redacted"
      "KeyType": "4096"
    },
    "Certificates": null
  }
}`

So I just get the certificates: null and no new certificate. Any thoughts?

**Comment by ktsenvolt** [comment]
I had similar issue before on my Raspberry Pi.


Traefik ssl lets-encrypt certificates not renewing

I have followed this guide to set up traefik on https://docs.teslamate.org/docs/guides/traefik and it worked, generating and renewing SSL certificates. But recently it has stopped working. keep getting emails about certificates expiring and forcing traefik to regenerate certificates in "acme.json" by deleting and touching the file does not work. Does anyone know how I can debug this?


https://doc.traefik.io/traefik/observability/logs/#level


Solution: Make sure ufw allow port 443, 80 on Raspberry p4

pi@raspberrypi:~ $ sudo ufw status
Status: Active

To                         Action      From
--                         ------      ----
80                         ALLOW       Anywhere                  
443                        ALLOW       Anywhere                  
22                         ALLOW       Anywhere                  
80 (v6)                    ALLOW       Anywhere (v6)             
443 (v6)                   ALLOW       Anywhere (v6)             
22 (v6)                    ALLOW       Anywhere (v6)

**Reply by trezn0** [reply]
I have it hosted on Google cloud, and I have opened the firewall on these ports. But I do indeed get an error related to ports, so I guess I need to do more troubleshooting on the firewall.

time="2023-11-17T08:26:48Z" level=error msg="Unable to obtain ACME certificate for domains \"xxx\": unable to generate a certificate for the domains [xxx]: error: one or more domains had a problem:\n[xxx] acme: error: 400 :: urn:ietf:params:acme:error:connection :: 1.2.3.4: Fetching http://xxx/.well-known/acme-challenge/omfV6P7vdNv4aPxOwjjWmJ1CIz5_ePs0F0ssJFX4AKA: Timeout during connect (likely firewall problem)\n" rule="Host(`xxx`) && (Path(`/grafana`) || PathPrefix(`/grafana/`))" routerName=grafana@docker providerName=tmhttpchallenge.acme ACME CA="https://acme-v02.api.letsencrypt.org/directory"

**Comment by trezn0** [comment]
Fixed the issue. I had disabled http traffic in the VM instance firewall. In the overall firewall port 80 is open, I didn't think if the extra layer there. So it works again after enabling this traffic.
